corehq/ex-submodules/casexml/apps/case/sharedmodels.py

```diff
@@ -45,11 +45,15 @@ def from_case_index_update(cls, index):
                    relationship=index.relationship,)
 
     def __unicode__(self):
-        return "%(identifier)s ref: (type: %(ref_type)s, id: %(ref_id)s)" % \
-                {"identifier": self.identifier,
-                 "ref_type": self.referenced_type,
-                 "ref_id": self.referenced_id}
-    
+        return unicode(
+            "CommCareCaseIndex("
+            "identifier='{index.identifier}', "
+            "referenced_type='{index.referenced_type}', "
+            "referenced_id='{index.referenced_id}', "
+            "relationship='{index.relationship}'"
+            ")"
+        ).format(index=self)
```

Comment:
why not use unicode literals here rather than calling `unicode(...)`?

haha, before it wasn't even returning a unicode string